flink-architecture-tests/flink-architecture-tests-test/src/main/java/org/apache/flink/architecture/rules/ITCaseRules.java

```diff
@@ -98,23 +99,28 @@
     @ArchTest
     public static final ArchRule ITCASE_USE_MINICLUSTER =
             freeze(
-                    javaClassesThat()
-                            .haveSimpleNameEndingWith("ITCase")
-                            .and()
-                            .areTopLevelClasses()
-                            .and()
-                            .doNotHaveModifier(ABSTRACT)
-                            .should(
-                                    fulfill(
-                                            // JUnit 5 violation check
-                                            miniClusterExtensionRule()
-                                                    .and(allCallbackWrapper())
-                                                    // JUnit 4 violation check, which should be
-                                                    // removed
-                                                    // after the JUnit 4->5 migration is closed.
-                                                    // Please refer to FLINK-25858.
-                                                    .or(miniClusterWithClientResourceClassRule())
-                                                    .or(miniClusterWithClientResourceRule()))));
+                            javaClassesThat()
+                                    .haveSimpleNameEndingWith("ITCase")
+                                    .and()
+                                    .areTopLevelClasses()
+                                    .and()
+                                    .doNotHaveModifier(ABSTRACT)
+                                    .should(
+                                            fulfill(
+                                                    // JUnit 5 violation check
+                                                    miniClusterExtensionRule()
+                                                            .and(allCallbackWrapper())
+                                                            // JUnit 4 violation check, which should
+                                                            // be
+                                                            // removed
+                                                            // after the JUnit 4->5 migration is
+                                                            // closed.
+                                                            // Please refer to FLINK-25858.
+                                                            .or(
+                                                                    miniClusterWithClientResourceClassRule())
+                                                            .or(
+                                                                    miniClusterWithClientResourceRule()))))
+                    .as("ITCASE tests should define MiniCluster explicitly");
```

Comment:
```suggestion
                    .as("ITCASE tests should use a MiniClusterWithClientResource or MiniClusterExtension");
```

Reply:
I was actually struggling to this. If we used the concrete class name, the rule key will change anytime when the class name has been refactored.

Reply:
I don't think we'll have a good reason to rename this class anytime soon, especially since this is "kind of" an API.

Reply:
we can generalize it to say something along the lines of "should use a MiniCluster resource or extension"